src/components/Dropdown/Dropdown.vue

```diff
@@ -117,13 +117,21 @@
                 :placeholder="item.label === placeholder"
                 @mousedown="onOptionMousedown"
                 @mouseenter="onOptionMouseover"
-                @click="onOptionClick"
+                @click="($event, index) => confirmChangeModal ? tryOnOptionClick($event, index) : onOptionClick($event, index)"
```

Comment:
I would change this line (120) and the above (104) into a method, given it's long and exactly the same.